src/clj/rems/db/fix_userid.clj

```diff
@@ -0,0 +1,346 @@
+     (do
+       (apply prn #'fix-apikey api-key params)
+       (when-not simulate?
+         (apply rems.db.api-key/update-api-key! params))
+       {:api-key api-key :params params}))))
+
+(comment
+  (fix-apikey "alice" "charlie" false))
+
+(defn fix-application-event [old-userid new-userid simulate?]
+  (doall
+   (for [old-event (rems.db.events/get-all-events-since 0)
+         :let [new-event (cond-> old-event
+                           (= old-userid (:event/actor old-event))
+                           (assoc :event/actor new-userid)
+
+                           (contains? (set (:application/reviewers old-event)) old-userid)
+                           (update :application/reviewers #(replace {old-userid new-userid} %))
+
+                           (contains? (set (:application/deciders old-event)) old-userid)
+                           (update :application/deciders #(replace {old-userid new-userid} %))
+
+                           (= old-userid (:application/member old-event))
+                           (assoc :application/member new-userid)
+
+                           (= old-userid (:application/applicant old-event))
+                           (assoc :application/applicant new-userid))]
+         :when (not= new-event old-event)
+         :let [params [new-event]]]
+     (do
+       (apply prn #'fix-application-event old-event params)
+       (when-not simulate?
+         (apply rems.db.events/update-event! params))
+       {:old-event old-event :params params}))))
+
+(comment
+  (fix-application-event "alice" "frank" false)
+  (fix-application-event "carl" "charlie" true))
+
+(defn fix-attachment [old-userid new-userid simulate?]
+  (doall
+   (for [attachment (rems.db.attachments/get-attachments)
+         :when (= old-userid (:attachment/user attachment))
+         :let [params [(assoc attachment :attachment/user new-userid)]]]
+     (do
+       (apply prn #'fix-attachment attachment params)
+       (when-not simulate?
+         (apply rems.db.attachments/update-attachment! params))
+       {:attachment attachment :params params}))))
+
+(comment
+  (fix-attachment "alice" "frank" false))
+
+(defn fix-audit-log [old-userid new-userid simulate?]
+  (doall
+   (for [audit-log (rems.db.core/get-audit-log)
+         :when (= old-userid (:userid audit-log))
+         :let [params [(merge audit-log
+                              {:time-new (:time audit-log)
+                               :path-new (:path audit-log)
+                               :method-new (:method audit-log)
+                               :apikey-new (:apikey audit-log)
+                               :userid-new new-userid
+                               :status-new (:status audit-log)})]]]
+     (do
+       (apply prn #'fix-audit-log audit-log params)
+       (when-not simulate?
+         (let [result (apply rems.db.core/update-audit-log! params)]
+           (assert (= 1 (first result)) {:audit-log audit-log :params params :result result})))
```

Comment:
currently the audit log structure expects a certain format, but nevertheless the rename functionality seems like something that should be contained in audit log. what do you think?

Reply:
Do you mean that it should be recorded in the audit log that this script was run? I kind of support that notion but it's also limited to requests to the server in practice. Also we decided to rename the user in the audit log as well.

Reply:
We don't record database migrations, which could be one way to think about this. Also other CLI things may pass the audit log like granting roles, not sure.

Reply:
Agreed not to add lines to the table. Also noticed that it wouldn't record in detail what it did and that's perhaps more important to save for later questions.